src/main/sensors/battery.c

```diff
@@ -545,7 +552,7 @@ void currentMeterUpdate(timeUs_t timeDelta)
                 break;
             }
         case CURRENT_SENSOR_VIRTUAL:
-            amperage = batteryMetersConfig()->current.offset;
+            amperage = currentBatteryProfile->batteryMeters.current.offset;
```

Comment:
Maybe instead of changing the current scale to the battery profile, adding a second scale to the battery profile that can be used for the virtual sensor is a better solution.

It would not required as many changes and would only affect users of the virtual current sensor.